service/history/historyEngine.go

```diff
@@ -1996,323 +1957,6 @@ func (s *shardContextWrapper) NotifyNewHistoryEvent(event *historyEventNotificat
-	if attributes.GetExecutionStartToCloseTimeoutSeconds() <= 0 {
-		attributes.ExecutionStartToCloseTimeoutSeconds = common.Int32Ptr(executionInfo.WorkflowTimeout)
-	}
-
-	// Inherit decision task timeout from previous execution if not provided on decision
-	if attributes.GetTaskStartToCloseTimeoutSeconds() <= 0 {
-		attributes.TaskStartToCloseTimeoutSeconds = common.Int32Ptr(executionInfo.DecisionTimeoutValue)
-	}
-
-	return nil
-}
-
-func validateStartChildExecutionAttributes(parentInfo *persistence.WorkflowExecutionInfo,
-	attributes *workflow.StartChildWorkflowExecutionDecisionAttributes, maxIDLengthLimit int) error {
-	if attributes == nil {
-		return &workflow.BadRequestError{Message: "StartChildWorkflowExecutionDecisionAttributes is not set on decision."}
-	}
-
-	if attributes.GetWorkflowId() == "" {
-		return &workflow.BadRequestError{Message: "Required field WorkflowID is not set on decision."}
-	}
-
-	if attributes.WorkflowType == nil || attributes.WorkflowType.GetName() == "" {
-		return &workflow.BadRequestError{Message: "Required field WorkflowType is not set on decision."}
-	}
-
-	if attributes.ChildPolicy == nil {
-		return &workflow.BadRequestError{Message: "Required field ChildPolicy is not set on decision."}
-	}
-
-	if len(attributes.GetDomain()) > maxIDLengthLimit {
-		return &workflow.BadRequestError{Message: "Domain exceeds length limit."}
-	}
-
-	if len(attributes.GetWorkflowId()) > maxIDLengthLimit {
-		return &workflow.BadRequestError{Message: "WorkflowId exceeds length limit."}
-	}
-
-	if len(attributes.WorkflowType.GetName()) > maxIDLengthLimit {
-		return &workflow.BadRequestError{Message: "WorkflowType exceeds length limit."}
-	}
-
-	if err := common.ValidateRetryPolicy(attributes.RetryPolicy); err != nil {
-		return err
-	}
-
-	if err := cron.ValidateSchedule(attributes.GetCronSchedule()); err != nil {
-		return err
-	}
-
-	// Inherit tasklist from parent workflow execution if not provided on decision
-	if attributes.TaskList == nil || attributes.TaskList.GetName() == "" {
-		attributes.TaskList = &workflow.TaskList{Name: common.StringPtr(parentInfo.TaskList)}
-	}
-
-	if len(attributes.TaskList.GetName()) > maxIDLengthLimit {
-		return &workflow.BadRequestError{Message: "TaskList exceeds length limit."}
-	}
-
-	// Inherit workflow timeout from parent workflow execution if not provided on decision
-	if attributes.GetExecutionStartToCloseTimeoutSeconds() <= 0 {
-		attributes.ExecutionStartToCloseTimeoutSeconds = common.Int32Ptr(parentInfo.WorkflowTimeout)
-	}
-
-	// Inherit decision task timeout from parent workflow execution if not provided on decision
-	if attributes.GetTaskStartToCloseTimeoutSeconds() <= 0 {
-		attributes.TaskStartToCloseTimeoutSeconds = common.Int32Ptr(parentInfo.DecisionTimeoutValue)
-	}
-
-	return nil
-}
-
 func validateStartWorkflowExecutionRequest(request *workflow.StartWorkflowExecutionRequest, maxIDLengthLimit int) error {
```

Comment:
Can this also move to the new file?

Reply:
this has nothing related to decision validation